pymatgen/io/lobster/inputs.py

```diff
@@ -182,6 +184,7 @@ def diff(self, other):
         similar_param = {}
         different_param = {}
         key_list_others = [element.lower() for element in other]
+        new_key = None
```

Comment:
```suggestion
        new_key = None
```

I am a bit worried about those. I think we should rather name them differently

Reply:
You could just push your fixes here, no problem at all and I would deeply appreciate your help!

Currently I'm just trying to assign some of the possibly undefined variables with a default `None` value to prevent variables from being unbound (which doesn't alter the code logic by any means), which means if the code is designed in a way that could exit the loop without being assigned at all, the code would still fail.

For example:
```python
idx = None  # added in this PR

for idx in range(value):
    if idx % 2 == 0:
        do something

print(lines[index])  # might throw an error 

```

If is an irrational is assigned to `value`(for example `-1`), the loop would not be entered and `idx` would still be `None`. If we don't assign `idx = None`, it would then become a `NameError`. 

Seemingly this doesn't improve the robustness of the code at all, but my hope is this would be the first step, afterwards we would add type annotations and checks to ensure `idx` is a `int`, if not the code should exist gracefully. Sorry it's just too much work 😅, so I'm not planning to refine the code logic at least in this PR .

Reply:
Thanks for fixing. Things look much cleaner now :)

Reply:
Yeah. This was written a very long time ago without thinking too much about the coding style 🙃. We will find similar things all over the code.
Once this is merged, I will check if any of this breaks LobsterPy (e.g., renaming of some ValueError messages). 
Thanks for working on these issues!

Reply:
Maybe, we have to go through this one as well. I guess the initialisation here with None is probably also not necessary.